File: Unit 2-Data Structure/Stack/As LinkedList/Stack.java
class Node<T>
{
    T data;
    Node<T> next;

    public Node(T data)
    {
        this.data = data;
        this.next = null;
    }
}

class LStack<T>
{
    private Node<T> top;

    public LStack()
    {
        this.top = null;
    }

    public void push(T value)
    {
        Node<T> newNode = new Node<>(value);
        newNode.next = top;
        top = newNode;
    }

    public T pop()
    {
        if (isEmpty())
        {
            System.out.println("Stack Underflow");
            return null;
        } else {
            T item = top.data;
            top = top.next;
            return item;
        }
    }

    public boolean isEmpty()
    {
        return top == null;
    }

    public T peek()
    {
        if (isEmpty()) {
            System.out.println("Stack Underflow");
            return null;
        } else {
            return top.data;
        }
    }

}

class Stack
{
    public static void main(String[] args)
    {
        LStack<Integer> myStack = new LStack<>();

        for (int i = 0; i <= 10; i++)
        {
            myStack.push(i);
        }

        System.out.println("Top element: " + myStack.peek());
        System.out.println("Is the stack empty? " + (myStack.isEmpty() ? "Yes" : "No"));
    }
}

File: Unit 2-Data Structure/Stack/As LinkedList/Stack.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

class LStack:
    def __init__(self):
        self.top = None

    def push(self, value):
        new_node = Node(value)
        new_node.next = self.top
        self.top = new_node

    def pop(self):
        if self.is_empty():
            print("Stack Underflow")
            return None
        else:
            item = self.top.data
            self.top = self.top.next
            return item

    def is_empty(self):
        return self.top is None

    def peek(self):
        if self.is_empty():
            print("Stack Underflow")
            return None
        else:
            return self.top.data

if __name__ == "__main__":
    my_stack = LStack()

    for i in range(11):
        my_stack.push(i)

    print("Top element:", my_stack.peek())
    print("Is the stack empty?", "Yes" if my_stack.is_empty() else "No")
